src/command.ts

```diff
@@ -29,104 +36,116 @@ export type CypressConfigEnv = {
+          // Screenshot already taken
+          visualRegressionOptions.screenshotAbsolutePath = screenShotProps.path
+          visualRegressionOptions.spec = Cypress.spec
+          switch (visualRegressionOptions.type) {
+            case 'regression':
+              return compareScreenshots(subject, visualRegressionOptions)
+            case 'base':
+              return updateSnapshots(subject, visualRegressionOptions)
+            default:
+              throw new Error(
+                `The 'type' environment variable is invalid. Expected: 'regression' or 'base' instead of '${visualRegressionOptions.type}'`
+              )
+          }
         }
-      })
+      )
     }
   )
 }
 
 function prepareOptions(
   name: string,
-  errorThreshold: number,
-  screenshotOptions: ScreenshotOptions
+  screenshotOptions: Partial<Cypress.ScreenshotOptions & PluginOptions>,
+  commandOptions: Partial<Cypress.ScreenshotOptions & PluginOptions>
 ): VisualRegressionOptions {
-  if (Cypress.env('visualRegression') !== undefined) {
-    throw new Error(
-      'Environment variables under "visualRegression" object (Version 4) is deprecated, please use single keys, i.e. visualRegressionType, visualRegressionBaseDirectory, etc.'
-    )
+  // if type is not set, throw error immediately
+  if (Cypress.env('visualRegressionType') === undefined) {
+    throw new Error("The 'type' environment variable is missing. Expected values: 'regression' or 'base'")
+  }
+
+  // follow Cypress priority order https://docs.cypress.io/guides/references/configuration#Resolved-Configuration
+  // firstly, set default values
+  const defaultCommandOptions = {
+    errorThreshold: 0,
+    failSilently: false,
+    pixelmatchOptions: { threshold: 0.1 }
   }
+
   const options: VisualRegressionOptions = {
     type: Cypress.env('visualRegressionType'),
     screenshotName: name,
-    specName: Cypress.spec.name,
     screenshotAbsolutePath: 'null', // will be set after takeScreenshot
-    screenshotOptions,
-    errorThreshold,
-    baseDirectory: Cypress.env('visualRegressionBaseDirectory'),
-    diffDirectory: Cypress.env('visualRegressionDiffDirectory'),
-    generateDiff: Cypress.env('visualRegressionGenerateDiff'),
-    failSilently: Cypress.env('visualRegressionFailSilently'),
+    pluginOptions: defaultCommandOptions,
+    screenshotOptions: {},
+    baseDirectory: 'cypress/snapshots/base',
+    diffDirectory: 'cypress/snapshots/diff',
+    generateDiff: 'fail',
     spec: Cypress.spec
   }
 
-  if (screenshotOptions?.failSilently !== undefined) {
-    options.failSilently = screenshotOptions.failSilently
+  // secondly, override values provided in e2e support file
+  if (screenshotOptions.pixelmatchOptions !== undefined) {
+    options.pluginOptions.pixelmatchOptions = screenshotOptions.pixelmatchOptions
+  }
+  // TODO refactor this in next release, separate plugin vs screenshot properties
+  options.screenshotOptions = { ...screenshotOptions, ...commandOptions }
```

Comment:
This annoys me, and would like to change it in the next breaking change release (as this will also be a breaking change)
It will make it much easier to maintain this if we split the options up:
```
addCompareSnapshotCommand({
  pluginOptions: {
    errorThreshold: 0,
    failSilently: false,
  },
  pixelmatchOptions: {
    threshold: 0.1
  },
  screenshotOptions: {
    capture: 'fullPage'
  }
})
```